test/coveralls.test.js

```diff
@@ -3,19 +3,80 @@
 const Code = require('code')
 const Lab = require('lab')
 const Seneca = require('seneca')
+var _ = require('lodash')
+
+const Proxyquire = require('proxyquire')
+const NpmSenecaFakeData = require('./npm-data')
+const CoverallsSenecaFakeData = require('./coveralls-data')
+const InvalidPluginName = 'qwerty_qwerty'
 
 const lab = exports.lab = Lab.script()
 const describe = lab.describe
-const suite = lab.suite
 const before = lab.before
 const it = lab.it
 const expect = Code.expect
 
+const NpmRegistry = 'http://registry.npmjs.org/'
+
+const RequestMap = [
+  {
+    urlMatch: NpmRegistry + 'seneca',
+    err: null,
+    response: {},
+    body: JSON.stringify(NpmSenecaFakeData)
+  },
+  {
+    urlMatch: InvalidPluginName,
+    err: null,
+    response: {},
+    body: '{}'
+  },
+  {
+    urlMatch: 'coveralls.io',
+    err: null,
+    response: {},
+    body: JSON.stringify(CoverallsSenecaFakeData)
+  }
+]
+
+// const DefaultRequestMap = [
+//   {
+//     urlMatch: NpmRegistry + 'seneca',
+//     err: null,
+//     response: {},
+//     body: JSON.stringify(NpmSenecaFakeData)
+//   },
+//   {
+//     urlMatch: 'coveralls.io',
+//     err: null,
+//     response: {},
+//     body: JSON.stringify(CoverallsSenecaFakeData)
+//   }
+// ]
+
+const RequestProxy = {
+  request: {
+    get: (opts, done) => {
+      const request = _.find(RequestMap, (o) => {
```

Comment:
Thats fair 😊. consistently is the primary reason.

Kindest Regards,

Dean

> On 15 Jun 2016, at 17:42, Cristian Kiss <notifications@github.com> wrote:
> 
> In test/coveralls.test.js:
> 
> > +//     err: null,
> > +//     response: {},
> > +//     body: JSON.stringify(NpmSenecaFakeData)
> > +//   },
> > +//   {
> > +//     urlMatch: 'coveralls.io',
> > +//     err: null,
> > +//     response: {},
> > +//     body: JSON.stringify(CoverallsSenecaFakeData)
> > +//   }
> > +// ]
> > +
> > +const RequestProxy = {
> > +  request: {
> > +    get: (opts, done) => {
> > +      const request = _.find(RequestMap, (o) => {
> @mcdonnelldean cool, but can you please explain? I'm just curious :)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>